utils/dbt_docs/dbt_classes_and_funcs.py

```diff
@@ -0,0 +1,783 @@
+        self.original_file_path = re.sub('bigquery|snowflake|default|redshift\_postgres|redshift|postgres|databricks', '&lt;adaptor&gt;', self.original_file_path)
+        # ensure all column names are lowercase
+        temp_cols = deepcopy(self.columns)
+        self.columns = {k.lower(): v for k, v in temp_cols.items()}
+
+    def to_markdown(self, key, docs):
+        """Generates the markdown for a dbt model
+
+        Args:
+            key (str): Full name of the model (key from master dict)
+            docs (dict): Dictionary of docs objects for lookup
+
+        Returns:
+            str: A markdown string for the model
+        """
+        # Get all the information we need about the macro
+        description = get_doc(self.description, docs, key) # If this was a disabled model we need to process from the actual docs
+        dispatched_sql = self.dispatched_sql
+        dispatched_filepath = self.paths
+        columns = self.columns
+        # Remove all integration test stuff from depends/refs, also de-dupe in case any made it through somehow
+        depends_macros = list({x for x in self.depends_on.get('macros') if 'integration_tests' not in x})
+        depends_models = list({x for x in self.depends_on.get('nodes') if 'integration_tests' not in x})
+        referenced_by_macros = list({x for x in self.referenced_by.get('macros') if 'integration_tests' not in x})
+        referenced_by_models = list({x for x in self.referenced_by.get('nodes') if 'integration_tests' not in x})
+
+        # Set initial markdown, set specific header key
+        markdown = [f'### {self.name.replace("_"," ").title()} {{#{key}}}',
+                    '',
+                    '<DbtDetails><summary>',
+                    f'<code>{self.original_file_path}</code>',
+                    '</summary>',
+                    '',
+                    '#### Description']
+
+        # Add the description
+        if description != "":
+            is_documented = True
+            markdown.append(description)
+        else:
+            is_documented = False
+            markdown.append('This model does not currently have a description.')
+
+        if self.type is not None:
+            markdown.extend(['', f'**Type**: {self.type}'])
+        # Add the file paths if it is adaptor split
+        if '&lt;adaptor&gt;' in self.original_file_path:
+            markdown.extend(['', '#### File Paths'])
+            markdown.append('<Tabs groupId="dispatched_sql">')
+            for lang in sorted(dispatched_filepath):
+                markdown.extend([f'<TabItem value="{lang}" label="{lang}" {"default" if lang == "default" else ""}>',
+                            '',
+                            f'`{dispatched_filepath[lang]}`',
+                            ''
+                            '</TabItem>'
+                            ])
+            markdown.extend(['</Tabs>', ''])
+
+        markdown.extend(['', '#### Details'])
```

Comment:
Switch headers to HTML so that they don’t produce permalink buttons (which then don’t work properly — discussed on slack)